docs: fix linting errors in angular package format guide (#47342)
angular/angular: aio/content/guide/angular-package-format.md

Most libraries with a single logical purpose should be published as a single entrypoint.
`@angular/core` for example uses a single entrypoint for the runtime, because the Angular runtime is generally used as a single entity.

> ### Resolution of secondary entrypoints

Reviewer: In all the typescript documentation I could find that mentions the term, `entry point` seems to be two words.
I'll add that to the spell checker for future linting, but could you make that correction in the files you edited, please?

Author: Fixed this to read as "entry points"

</code-example>

## Partial compilation

Libraries in the Angular Package Format must be published in "partial compilation" mode.
This is a compilation mode for `ngc` which produces compiled Angular code that is not tied to a specific Angular runtime version, in contrast to the full compilation used for applications, where the Angular compiler and runtime versions must match exactly.

> To partially compile Angular code, use the `"compilationMode"` flag in the `"angularCompilerOptions"` property of your `tsconfig.json`:

Reviewer: ```suggestion
To partially compile Angular code, use the `compilationMode` flag in the `angularCompilerOptions` property of your `tsconfig.json`:
```
Code format variable and property names don't need double quotes around. The code format is enough notation.

## Optimizations

### Flattening of ES modules

The Angular Package Format specifies that code be published in "flattened" ES module format.
This significantly reduces the build time of Angular applications as well as download and parse time of the final application bundle.
Please check out the excellent post ["The cost of small modules"](https://nolanlawson.com/2016/08/15/the-cost-of-small-modules) by Nolan Lawson.

> The Angular compiler has support for generating index ES module files that can then be used to generate flattened modules using tools like Rollup, resulting in a file format called Flattened ES Module or FESM.

Reviewer: ```suggestion
The Angular compiler can generate index ES module files. Tools like Rollup can use these files to generate flattened modules in a *Flattened ES Module* (FESM) file format.
```

FESM is a file format created by flattening all ES Modules accessible from an entrypoint into a single ES Module.
It's formed by following all imports from a package and copying that code into a single file while preserving all public ES exports and removing all private imports.

> The shortened name "FESM" \(pronounced "phe-som"\) can have a number after it such as "FESM5" or "FESM2015" .

Reviewer: ```suggestion
The abbreviated name, FESM, pronounced *phe-som*, can be followed by a number such as FESM5 or FESM2015.
```

The number refers to the language level of the JavaScript inside the module.

> Accordingly a FESM5 file would be ESM+ES5 \(import/export statements and ES5 source code\).

Reviewer: ```suggestion
Accordingly a FESM5 file would be ESM+ES5 and include import/export statements and ES5 source code.
```

#### Note about the defaults in package.json

> As of webpack v4 the flattening of ES modules optimization should not be necessary for webpack users, and in fact theoretically it should be possible to get better code-splitting without flattening of modules in webpack.

Reviewer: ```suggestion
As of webpack v4, the flattening of ES modules optimization should not be necessary for webpack users. It should be possible to get better code-splitting without flattening of modules in webpack.
```

> In practice however size regressions are seen when using unflattened modules as input for webpack v4.

Reviewer: ```suggestion
In practice, size regressions can still be seen when using unflattened modules as input for webpack v4.
```

> This is why `"module"` and `"es2020"` package.json entries still point to FESM files.

Reviewer: ```suggestion
This is why `module` and `es2020` package.json entries still point to FESM files.
```

> This issue is being investigated and it is expected for a switch to be made to the `"module"` and `"es2020"` package.json entry points to unflattened files when the size regression issue is resolved.

Reviewer: ```suggestion
This issue is being investigated. It is expected to switch the `module` and `es2020` package.json entry points to unflattened files after the size regression issue is resolved.
```

Build tools such as Webpack support a flag which allows packages to declare that they do not depend on side-effectful code at the top level of their modules, giving the tools more freedom to tree-shake code from the package.
The end result of these optimizations should be smaller bundle size and better code distribution in bundle chunks after code-splitting.
This optimization can break your code if it contains non-local side-effects - this is however not common in Angular applications and it's usually a sign of bad design.
The recommendation is for all packages to claim the side-effect free status by setting the `sideEffects` property to `false`, and that developers follow the [Angular Style Guide](https://angular.io/guide/styleguide) which naturally results in code without non-local side-effects.

More info: [webpack docs on side-effects](https://github.com/webpack/webpack/tree/master/examples/side-effects)

### ES2020 language level

ES2020 Language level is now the default language level that is consumed by Angular CLI and other tooling.

> The Angular CLI should downlevel the bundle to a language level that is supported by all targeted browsers at application build time.

Reviewer: ```suggestion
The Angular CLI down-levels the bundle to a language level that is supported by all targeted browsers at application build time.
```
